rules/no-await-in-promise-methods.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const MESSAGE_ID = 'no-await-in-promise-methods';
+const messages = {
+	[MESSAGE_ID]: 'Promise.{{ method }} contains an await.',
+};
+
+const isPromiseMethodWithArray = node =>
+	node.callee.type === 'MemberExpression'
+	&& node.callee.object.type === 'Identifier'
+	&& node.callee.object.name === 'Promise'
+	&& node.callee.property.type === 'Identifier'
+	&& ['all', 'allSettled', 'any', 'race'].includes(node.callee.property.name)
```

Comment:
`['all', 'allSettled', 'any', 'race']` should be a top-level variable to make it easier to add more in the future.

Reply:
It's corrected.